### ISSUE TITLE
[CLI]: can't catch exception on torch.multiprocessing.spawn after wandb.init

### ISSUE THREAD

**Issue by MilkClouds** [wandb/wandb#4047]
### Describe the bug

On process spawned by torch.mp.spawn, after executing wandb.init, process DO NOT catch exception.
After executing below script, process hang and never die. Also it does not print exception. Do not execute wandb.finish().

```python
import wandb, torch
def worker(gpu, n_gpus):
    print("HI!")
    if gpu == 0:
        wandb.init(project='test')
        1//0
        wandb.finish()
    print("BYE!")
if __name__ == "__main__":
    torch.multiprocessing.spawn(worker, nprocs=2, args=(2, ))
```

```shell
(open-mmlab) PS D:\Users\MilkClouds\Documents\GitHub> python test.py
HI!HI!

BYE!
wandb: Currently logged in as: milkclouds (use `wandb login --relogin` to force relogin)
wandb: wandb version 0.12.21 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Tracking run with wandb version 0.12.9
wandb: Syncing run feasible-water-4
wandb:  View project at https://wandb.ai/milkclouds/test
wandb:  View run at https://wandb.ai/milkclouds/test/runs/3evff9i1
wandb: Run data is saved locally in D:\Users\MilkClouds\Documents\GitHub\wandb\run-20220803_225905-3evff9i1
wandb: Run `wandb offline` to turn off syncing.
```

I expected error message of division by zero and wandb got finished, but process hang and never die.

### Environment

WandB version: 0.12.9, 0.12.21 (Windows 11 and WSL2)

OS: Tested on Windows 11 ans WSL2

Python version: 3.7.11, 3.9.12 (Windows 11 and WSL2)

Versions of relevant libraries: 1.10.0, 1.12.0 (Windows 11 and WSL2)

**Comment by ShaohanTian** [comment]
I now meet this problem too, have you solved it?@MilkClouds

**Comment by MilkClouds** [comment]
@ShaohanTian I think it is solved in wandb version 0.13.7.
I could reproduce same phenomena with wandb 0.12.21, but right after upgrading wandb to 0.13.7, error message outcomes. As long as I think it is now solved, I'll close this issue.
